plugins/inputs/tail/tail.go

```diff
@@ -229,21 +268,18 @@ func (t *Tail) receiver(parser parsers.Parser, tailer *tail.Tail) {
 
 		for _, metric := range metrics {
 			metric.AddTag("path", tailer.Filename)
-			t.acc.AddMetric(metric)
 		}
-	}
-
-	t.Log.Debugf("Tail removed for %q", tailer.Filename)
 
-	if err := tailer.Err(); err != nil {
-		t.Log.Errorf("Tailing %q: %s", tailer.Filename, err.Error())
+		select {
+		case <-t.ctx.Done():
+			break
+		case t.sem <- empty{}:
```

Comment:
in a select, this won't block, it'll drop the metric. I assume that's what you wanted here.

Reply:
maybe a comment to that effect will make this obvious for anyone in the future less familiar with channel/select semantics.

Reply:
This should block because there is no default case.

Reply:
On 275 I should return, not break.